src/GraphQL/Validation/ValidationContext.cs

```diff
@@ -128,7 +128,10 @@ public List<GraphQLFragmentSpread> GetFragmentSpreads(GraphQLSelectionSet node)
 
         /// <summary>
         /// Returns all of the variable values defined for the operation from the attached <see cref="Variables"/> object.
+        /// Only correctly validated variables are returned. If the variable is specified incorrectly, then an instance of
+        /// <see cref="ValidationError"/> is added to the <see cref="Errors"/>.
         /// </summary>
+        [Obsolete("This method can repeatedly add the same error into ValidationContext.Errors collection when called multiple times. Use GetVariablesValuesAsync method instead.")]
         public async ValueTask<Variables> GetVariableValuesAsync(IVariableVisitor? visitor = null)
```

Comment:
Can this call the `GetVariablesValuesAsync` method?

Reply:
yes)